.github/workflows/test.yaml

```diff
@@ -0,0 +1,76 @@
+name: ID Mapping Service test
+
+on:
+  pull_request:
+    types:
+    - opened
+    - reopened
+    - synchronize
+    - ready_for_review
+  push:
+    # run workflow when merging to main or develop
+    branches:
+      - main
+      - master
+      - develop
+
+jobs:
+  id_mapping_service_tests:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - python: '3.9.19'
+            mongo: 'mongodb-linux-x86_64-3.6.23'
+            wired_tiger: 'false'
+
+    steps:
+    - name: Check out GitHub repo
+      uses: actions/checkout@v4
+
+    - name: Setup Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{matrix.python}}
+
+    - name: Install dependencies and set up test config
+      shell: bash
+      run: |
+
+        # TODO switch to pipenv
+        # install python libraries
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
+        pip install -r dev-requirements.txt
+
+        # set up mongo
+        cd ..
+        wget -q http://fastdl.mongodb.org/linux/${{matrix.mongo}}.tgz
+        tar xfz ${{matrix.mongo}}.tgz
+        export MONGOD=`pwd`/${{matrix.mongo}}/bin/mongod
+        cd -
+
+        # set up test config
+        cp -n test.cfg.example test.cfg
+        sed -i "s#^test.temp.dir =.*#test.temp.dir=temp_test_dir#" test.cfg
+        sed -i "s#^test.mongo.exe.*#test.mongo.exe=$MONGOD#" test.cfg
+        sed -i "s#^test.mongo.wired_tiger.*#test.mongo.wired_tiger=${{matrix.wired_tiger}}#" test.cfg
+
+    - name: Run tests
+      shell: bash
+      run: |
+        export HOMEDIR=`pwd`
+        export PYTHONPATH=$HOMEDIR/src:$PYTHONPATH
+        export IDMAP_TEST_FILE=$HOMEDIR/test.cfg
+
+        GITCOMMIT=$(git rev-parse HEAD)
+        echo "# Don't check this file into git please" > $HOMEDIR/src/jgikbase/idmapping/gitcommit.py
+        echo "commit = '${GITCOMMIT}'" >> $HOMEDIR/src/jgikbase/idmapping/gitcommit.py
+        pytest src/jgikbase/test --cov=src/jgikbase/idmapping --cov-report=xml
```

Comment:
You should use make test here to run all the tests that travis used to run

Reply:
It will run the all tests that travis used to run. But you prefer `make test`?

Reply:
It doesn't though: https://github.com/jgi-kbase/IDMappingService/blob/master/Makefile#L19-L22

Reply:
`flake8 src` failed. And it passed when using travis? We can either format or ignore them.

Reply:
If everything else is passing I'd say enable it and format in another PR

Reply:
Only Pytest is passing. If enable any other tests (flake8, mypy, bandit), GHA fails. 
So I disabled all three and will fix them in another PR.

Reply:
:+1: